frontend/src/context/auth/auth.tsx

```diff
@@ -0,0 +1,61 @@
+'use client';
+
+import {
+	createContext,
+	useState,
+	useContext,
+	useEffect,
+	ReactNode,
+} from 'react';
+import { configuration } from '@/config/Constants';
+
+interface AuthProviderProps {
+	children: ReactNode;
+}
+
+interface AuthContextType {
+	isAuthenticated: boolean;
+	loading: boolean;
+}
+
+const API_URL = configuration.url.API_URL;
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export function AuthProvider({ children }: AuthProviderProps) {
+	const [isAuthenticated, setIsAuthenticated] = useState<boolean>(false);
+	const [loading, setLoading] = useState<boolean>(true);
+
+	async function checkSession() {
```

Comment:
So I spoke about this previously but we may want to turn this into a [refresh flow](https://stackoverflow.com/questions/27726066/jwt-refresh-token-flow) instead of just a simple check. In that refresh, if we don't get a positive response, we can then either store that the user was trying to get somewhere (through a [Relay State](https://stackoverflow.com/questions/34350160/what-is-exactly-relaystate-parameter-used-in-sso-ex-saml)), redirect them to the login with no apparent state, or redirect the user to the front page (and delete the cookie). The latter is easiest, the former the hardest.